cmd/auth/auth.go

```diff
@@ -0,0 +1,239 @@
+package auth
+
+import (
+	"bytes"
+	"context"
+	"crypto/rand"
+	"crypto/sha256"
+	_ "embed"
+	"encoding/base64"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"net/http"
+	"net/url"
+	"os"
+	"os/signal"
+	"path"
+	"time"
+
+	"github.com/cli/browser"
+	constants "github.com/open-sauced/pizza-cli/pkg"
+	"github.com/spf13/cobra"
+)
+
+//go:embed success.html
+var successHTML string
+
+const loginLongDesc string = `Log into the application.
+
+This command initiates the GitHub auth flow to log you into the application by launching your browser`
+
+func NewLoginCommand() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "login",
+		Short: "Log into the CLI application via GitHub",
+		Long:  loginLongDesc,
+		Args:  cobra.NoArgs,
+		RunE: func(cmd *cobra.Command, args []string) error {
+			return run()
+		},
+	}
+
+	return cmd
+}
+
+func run() error {
+	codeVerifier, codeChallenge, err := pkce(constants.CodeChallengeLength)
+	if err != nil {
+		return fmt.Errorf("PKCE error: %v", err.Error())
+	}
+
+	supabaseAuthURL := fmt.Sprintf("https://%s.supabase.co/auth/v1/authorize", constants.SupabaseID)
+	queryParams := url.Values{
+		"provider":              {"github"},
+		"code_challenge":        {codeChallenge},
+		"code_challenge_method": {"S256"},
+		"redirect_to":           {"http://" + constants.AuthCallbackAddr + "/"},
+	}
```

Comment:
Question: I tried abit of an experiment where I logged out of my page on `insights.opensauced.pizza` and did `pizza login`. It automatically logged me in without opening the github auth flow. I suppose the same thing happens when hitting the "Connect with GitHub" button in the insights UI; it just automatically logs me back in.

Reply:
Yes. You won't see the GitHub auth page everytime and will log in automatically.